sources/visual/Trajectory.h

```diff
@@ -0,0 +1,24 @@
+#ifndef TRAJECTORY_H
+#define TRAJECTORY_H
+
+#include <vector>
+#include <memory>
+#include <iostream>
+#include "theglobals.h"
+#include <cmath>
+
+// abstract class as a base for every kind of trajectory
+class Trajectory {
+public:
+    virtual std::vector<double> computeTrajectory(double t, double initX, double initY){};
```

Comment:
Il est préférable de mettre cette méthode abstraite en pure virtual. ("= 0")
Celle-ci lève une alerte quand je compile, étant donnée que son implémentation ne retourne pas de valeur.

```suggestion
    virtual std::vector<double> computeTrajectory(double t, double initX, double initY) = 0;
```